### PR TITLE
Add support for implementing static members in derived interfaces.

### PR DESCRIPTION
**Pull request by AlekseyTs**
https://github.com/dotnet/csharplang/blob/main/proposals/static-abstracts-in-interfaces.md

Test plan: #60968

Tests in DefaultInterfaceImplementationTests.cs are adjusted only for methods. Tests for other members will be adjusted separately.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @jcouv, @dotnet/roslyn-compiler Please review

**Review by 333fred — APPROVED** [review]
LGTM (commit 2)

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler For the second review.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs [hunk 1]
```diff
@@ -987,7 +987,7 @@ protected void CheckFeatureAvailabilityAndRuntimeSupport(SyntaxNode declarationS
                     diagnostics.Add(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, location);
                 }
 
-                if (((!hasBody && IsAbstract) || IsVirtual) && IsStatic && !ContainingAssembly.RuntimeSupportsStaticAbstractMembersInInterfaces)
+                if (((!hasBody && IsAbstract) || IsVirtual) && !isExplicitInterfaceImplementation && IsStatic && !ContainingAssembly.RuntimeSupportsStaticAbstractMembersInInterfaces)
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs:990]: >isExplicitInterfaceImplementation [](http://example.com/codeflow?start=64&length=33)

Do we need the additional parameter given that we have `IsExplicitInterfaceImplementation` property (used in this method already)?

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 2)

**Comment by jcouv** [comment]
The spec probably needs updating to reflect decisions as opposed to options: https://github.com/dotnet/csharplang/blob/main/proposals/static-abstracts-in-interfaces.md#default-implementations

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs:990]: > Do we need the additional parameter given that we have ```IsExplicitInterfaceImplementation``` property (used in this method already)?

Good catch, I will follow up on this in a separate PR.

**Review by jcouv — APPROVED** [review]
LTGM Thanks (iteration 2). Extra parameter can be removed in later PR

**Pull request merged**